lib/app/utils/index_builder.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module Inferno
+  class IndexBuilder
+    @files = nil
+    @index = nil
+
+    # Prepares the IndexBuilder to start indexing files
+    # @return [nil]
```

Comment:
I removed the documentation for returning (not sure if that's what you meant), but I decided to continue to return nil because my editor was still inferring the return value, making it a bit confusing haha

Reply:
If we're going to bother explicitly returning something, we should make it something meaningful. I think there is a meaningful return value that can be used for non-failure results in these methods.

Reply:
For `IndexBuilder#build`, I could return a boolean that tells whether an index was generated (like I did before). Is that what you mean? Or maybe I could return the contents of the index/nil on failure? If I'm passing a block to `build`, would it still make sense to return these values?

Should I also add a boolean return value for `add_json_file` that tells whether the file was successfully appended or not?

Reply:
I think that it would make sense to return the contents of the index, and the contents of the json, respectively.

Reply:
Ok, I made it return the `contents` for both methods